src/structures/GuildChannel.js

```diff
@@ -571,7 +571,10 @@ class GuildChannel extends Channel {
    * @readonly
    */
   get deletable() {
-    return this.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_CHANNELS, false);
+    return (
+      this.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_CHANNELS, false) &&
+      (this.guild.rulesChannelID !== this.id || this.guild.publicUpdatesChannelID !== this.id)
```

Comment:
If the channel to delete is _either_ the rules channel or the public updates channel is enough.
It doesn't need to be _both_ 👀 
```suggestion
      this.guild.rulesChannelID !== this.id &&
      this.guild.publicUpdatesChannelID !== this.id
```

Reply:
oops, oversight